common/lib/xmodule/xmodule/modulestore/__init__.py

```diff
@@ -276,27 +277,49 @@ def _is_in_bulk_operation(self, course_key, ignore_case=False):
         return self._get_bulk_ops_record(course_key, ignore_case).active
 
 
-class ModuleStoreAssetInterface(object):
+class SortedAssetList(SortedListWithKey):
     """
-    The methods for accessing assets and their metadata
+    List of assets that is sorted based on an asset attribute.
     """
-    @contract(asset_list='SortedListWithKey', asset_id='AssetKey')
-    def _find_asset_in_list(self, asset_list, asset_id):
+    def __init__(self, **kwargs):
+        super(SortedAssetList, self).__init__(**kwargs)
```

Comment:
Or if it doesn't, does the binary search function need to use the provided sort key?